flytekit/common/translator.py

```diff
@@ -0,0 +1,260 @@
+            _identifier_model.ResourceType.WORKFLOW, settings.project, settings.domain, entity.name, settings.version
+        )
+
+        # Translate nodes
+        upstream_sdk_nodes = [
+            get_serializable(settings, n) for n in entity._nodes if n.id != _common_constants.GLOBAL_INPUT_NODE_ID
+        ]
+
+        cp_entity = SdkWorkflow(
+            nodes=upstream_sdk_nodes,
+            id=workflow_id,
+            metadata=entity.workflow_metadata.to_flyte_model(),
+            metadata_defaults=entity.workflow_metadata_defaults.to_flyte_model(),
+            interface=entity._interface,
+            output_bindings=entity._output_bindings,
+        )
+        # Reset just to make sure it's what we give it
+        cp_entity.id._project = settings.project
+        cp_entity.id._domain = settings.domain
+        cp_entity.id._name = entity.name
+        cp_entity.id._version = settings.version
+
+    elif isinstance(entity, Node):
+        if entity._flyte_entity is None:
+            raise Exception(f"Node {entity.id} has no flyte entity")
+
+        upstream_sdk_nodes = [
+            get_serializable(settings, n)
+            for n in entity._upstream_nodes
+            if n.id != _common_constants.GLOBAL_INPUT_NODE_ID
+        ]
+
+        if isinstance(entity._flyte_entity, PythonTask):
+            cp_entity = SdkNode(
+                entity._id,
+                upstream_nodes=upstream_sdk_nodes,
+                bindings=entity._bindings,
+                metadata=entity._metadata,
+                sdk_task=get_serializable(settings, entity._flyte_entity),
+            )
+            if entity._aliases:
+                cp_entity._output_aliases = entity._aliases
+        elif isinstance(entity._flyte_entity, Workflow):
+            cp_entity = SdkNode(
+                entity._id,
+                upstream_nodes=upstream_sdk_nodes,
+                bindings=entity._bindings,
+                metadata=entity._metadata,
+                sdk_workflow=get_serializable(settings, entity._flyte_entity),
+            )
+        elif isinstance(entity._flyte_entity, BranchNode):
+            cp_entity = SdkNode(
+                entity._id,
+                upstream_nodes=upstream_sdk_nodes,
+                bindings=entity._bindings,
+                metadata=entity._metadata,
+                sdk_branch=get_serializable(settings, entity._flyte_entity),
+            )
+        elif isinstance(entity._flyte_entity, LaunchPlan):
+            cp_entity = SdkNode(
+                entity._id,
+                upstream_nodes=upstream_sdk_nodes,
+                bindings=entity._bindings,
+                metadata=entity._metadata,
+                sdk_launch_plan=get_serializable(settings, entity._flyte_entity),
+            )
+        else:
+            raise Exception(f"Node contained non-serializable entity {entity._flyte_entity}")
+
+    elif isinstance(entity, LaunchPlan):
+        if entity._auth_role:
```

Comment:
nit: do you really need this check? don't you pass it along as is None or otherwise?

Reply:
+1, this is a registration time construct anyways